src/bit-systems/audio-zone-system.ts

```diff
@@ -0,0 +1,326 @@
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    if (isDebugEnabled) createMaterial();
+    defineQuery([AudioZone])(APP.world).forEach(zoneEid => {
+      if (isDebugEnabled) {
+        !debugObjects.has(zoneEid) && addZoneDebugObject(APP.world, zoneEid);
+      } else {
+        debugObjects.has(zoneEid) && releaseZoneDebugObject(APP.world, zoneEid);
+      }
+    });
+    if (!isDebugEnabled && debugMaterial) {
+      disposeMaterial(debugMaterial);
+      debugMaterial = null;
+    }
+  });
+  const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+  if (isDebugEnabled) createMaterial();
+});
+
+const addZoneDebugObject = (world: HubsWorld, zoneEid: number) => {
+  const geometry = new BoxGeometry();
+  const mesh = new Mesh(geometry, debugMaterial!);
+  const helper = new BoxHelper(mesh, DEBUG_BBAA_COLOR);
+
+  const parent = world.eid2obj.get(zoneEid) as Object3D;
+  parent.add(helper);
+  parent.updateMatrixWorld(true);
+
+  debugObjects.set(zoneEid, helper);
+};
+
+const releaseZoneDebugObject = (world: HubsWorld, zoneEid: number) => {
+  const helper = debugObjects.get(zoneEid)!;
+  helper?.removeFromParent();
+  disposeNode(helper);
+  debugObjects.delete(zoneEid);
+};
+
+const isZoneEnabled = (zoneEid: number): boolean => {
+  return Boolean(AudioZone.flags[zoneEid] & AUDIO_ZONE_FLAGS.ENABLED);
+};
+
+const getZoneBoundingBox = (zoneEid: number) => {
+  if (AudioZone.flags[zoneEid] & AUDIO_ZONE_FLAGS.DYNAMIC) {
+    const obj = APP.world.eid2obj.get(zoneEid)!;
+    return aabbs.get(zoneEid)!.setFromObject(obj);
+  } else {
+    return aabbs.get(zoneEid)!;
+  }
+};
+
+const zoneContains = (zoneEid: number, position: Vector3) => {
+  return getZoneBoundingBox(zoneEid).containsPoint(position);
+};
+
+const getEmitterParams = (zoneEid: number) => {
+  return APP.audioOverrides.get(zoneEid);
+};
+
+type AnyPredicate = (zoneEid: number) => boolean;
+const any = (set: Set<number>, predicate: AnyPredicate) => {
+  for (const item of set) {
+    if (predicate(item)) return true;
+  }
+  return false;
+};
+
+const isUpdated = (currZones: Set<number>, prevZones: Set<Number>) => {
+  return (
+    currZones && prevZones && (currZones.size !== prevZones.size || any(currZones, zoneEid => !prevZones.has(zoneEid)))
+  );
+};
+
+const getEmitterPosition = (() => {
+  const pos = new Vector3();
+  return (emitterEid: number) => {
+    const audio = APP.audios.get(emitterEid);
+    if (audio) {
+      audio.getWorldPosition(pos);
+    } else {
+      pos.set(0, 0, 0);
+    }
+    return pos;
+  };
+})();
+
+const applyEmitterParams = (emitterEid: number, params: AudioSettings) => {
+  APP.zoneOverrides.set(emitterEid, params);
+  const audio = APP.audios.get(emitterEid);
+  if (audio) {
+    updateAudioSettings(emitterEid, audio);
+  }
+};
+
+const restoreEmitterParams = (emitterEid: number) => {
+  APP.zoneOverrides.delete(emitterEid);
+  const audio = APP.audios.get(emitterEid);
+  if (audio) {
```

Comment:
If there is no audio, is it a bug?

Reply:
Yeah this is called as part of a query so if the audio eid is there the audio element should be in the store too, otherwise there is a bug somewhere else.